kotlinpoet/src/main/java/com/squareup/kotlinpoet/TypeName.kt

```diff
@@ -288,9 +288,28 @@ class ClassName internal constructor(
    * Returns a class name created from the given parts. For example, calling this with package name
    * `"java.util"` and simple names `"Map"`, `"Entry"` yields `Map.Entry`.
    */
+  @Deprecated("", level = DeprecationLevel.HIDDEN)
```

Comment:
This is a binary incompatible change that will affect Java callers:
```java
String[] names = { "simple2", "simple3" };
ClassName cn3 = new ClassName("package", "simple", names);
```
This code compiles before the change and won't compile after. Normally we would require a major version change to merge binary-incompatible code, and given that this change is an API enhancement, not a bug fix, I would rather postpone merging this until we're ready for 2.0. @JakeWharton what are your thoughts?

Reply:
It's source incompatible for that specific case, not binary. And I highly doubt anyone is using it that way. It's far more common to do `get("package", "Map", "Entry")` which remains source compatible.

I don't have any problem breaking source compatibility for a usability improvement.

Reply:
Yeah, that's fair.